### ISSUE TITLE
Incorrect dtype casting after moving model to XLA device

### ISSUE THREAD

**Issue by YangFei1990** [pytorch/xla#4167]
## 🐛 Bug

If I cast the model to the right dtype before moving it to XLA device, the casting is correct and the model output has right dtype. However, if I first cast the model to XLA device then cast the model to the dtype, the model output still hold the original dtype despite the model parameters already have the right dtype.
Not sure if this is a designed behavior or a bug, the example below could well explain the issue. If such behavior is as designed, can we have it documented somewhere? 
Thanks a lot for the help!

## To Reproduce
Here is the script to reproduce the issue
```
import torch
import torch_xla.core.xla_model as xm
import torch_xla.distributed.xla_multiprocessing as xmp
import torch.nn as nn

class TestModule(torch.nn.Module):
    def __init__(self):
        super().__init__()
        self.embedding = nn.Embedding(10, 3)
    
    def forward(self, input):
        x = self.embedding(input)
        print(x.dtype)
        return x

def run():
    device = xm.xla_device()
    model = TestModule()
    # cast here, everything will be right
    # model.to(torch.bfloat16)
    model.to(device)
    # cast here, the parameters will have bfloat16 but output will be float32
    model.to(torch.bfloat16)
    for name, p in model.named_parameters():
        print(f"name {name}, dtype {p.dtype}")
    dummy_data = torch.LongTensor([[1,2,4,5],[4,3,2,9]]).to(device)
    output = model(dummy_data)
    xm.mark_step()
    print(output.dtype)

def _mp_fn(index):
    run()

if __name__ == "__main__":
    xmp.spawn(_mp_fn, args=(), nprocs=1)
```
Running with `export GPU_NUM_DEVICES=1 && python test_dtype.py`

If dtype cast before moving to XLA device, I see prints
```
name embedding.weight, dtype torch.bfloat16
torch.bfloat16
torch.bfloat16
```
If dtype cast after moving to XLA device, I see prints
```
name embedding.weight, dtype torch.bfloat16
torch.float32
torch.float32
```

## Expected behavior

The output dtype will be the parameter dype regardless of casting before or after model moving to XLA device.

## Environment

 - Reproducible on XLA backend [CPU/TPU]: GPU
 - torch_xla version: 1.12

**Comment by JackCaoG** [comment]
@alanwaketan can you take a look?

**Comment by alanwaketan** [comment]
Sure.

**Comment by alanwaketan** [comment]
Yea, it looks like we don't handle:
https://github.com/pytorch/pytorch/blob/5eecfcf5f3d118a9e2d502dfb8689018c9591662/aten/src/ATen/native/native_functions.yaml#L6795-L6798

This should actually be a feature request.

P.S. TorchScript backend on the other hand seems supporting `_to_copy`:
https://github.com/pytorch/pytorch/blob/5eecfcf5f3d118a9e2d502dfb8689018c9591662/torch/csrc/lazy/ts_backend/ts_native_functions.cpp#L140-L270

Not sure if that's enough to support all the TensorConversions:
https://github.com/pytorch/pytorch/blob/5eecfcf5f3d118a9e2d502dfb8689018c9591662/aten/src/ATen/native/TensorConversions.cpp

**Comment by YangFei1990** [comment]
Hi @alanwaketan thanks a lot for the help! I'm new to Pytorch/XLA and very interested in its mechanism, could you elaborate more about the concepts so that I can understand better of the issue?
- What is the rootcause of the issue? What is that handle and how it works?
- How does TorchScript relate to this? My understanding is that TorchScript and XLA are different compilers.

**Comment by YangFei1990** [comment]
Thanks a lot for the explanation @alanwaketan! I still have a few follow-up questions to understand it better.
When you say that  `.to(dtype)` is not implemented and it can not be handled, what does it mean exactly? In the view of XLA, what does this op do? My understanding is that such unsupported op will break the graph (like `mark_step`), but if so it should not result in an incorrect dype. 
Also, checking the print, why does the parameters already have the right dtype but the output tensor has wrong dtype?

**Comment by alanwaketan** [comment]
Good catch. I missed that part. Actually, I was not able to reproduce the error with nightly. Can you try nightly again?
